src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExportClient/BaseOtlpHttpExportClient.cs

```diff
@@ -26,13 +26,16 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClie
     /// <typeparam name="TRequest">Type of export request.</typeparam>
     internal abstract class BaseOtlpHttpExportClient<TRequest> : IExportClient<TRequest>
     {
-        protected BaseOtlpHttpExportClient(OtlpExporterOptions options, HttpClient httpClient)
+        protected BaseOtlpHttpExportClient(OtlpExporterOptions options, HttpClient httpClient, string signalPath)
         {
             Guard.ThrowIfNull(options);
             Guard.ThrowIfNull(httpClient);
+            Guard.ThrowIfNull(signalPath);
             Guard.ThrowIfInvalidTimeout(options.TimeoutMilliseconds);
 
-            this.Endpoint = new UriBuilder(options.Endpoint).Uri;
```

Comment:
This line was intentionally making a copy of `options.Endpoint` to prevent later changes to the options from affecting the exporter after it's been constructed. Though, it might have not been necessary since I think Uri may be immutable in practice. Maybe safer to just make a copy?

Reply:
🤔 I always thought that `Uri` is immutable, see [`UriBuilder` remarks](https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder?view=net-6.0#remarks). Let me see if there is anything to the contrary.

Reply:
Looking at its source code some fields are filled on demand, but the calling code cannot change what the `Uri` points to. I think it is safe to remove the copy, that said, for everyone peace of mind near a release, for something that shouldn't be in a hot path, I will use the old code.